src/taipy/core/_repository/_v2/_filesystem_repository.py

```diff
@@ -0,0 +1,143 @@
+# the License. You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+# an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+
+import json
+import pathlib
+import shutil
+from typing import Any, Iterable, Iterator, List, Optional, Type, Union
+
+from src.taipy.core.common.typing import Entity, Json, ModelType
+from taipy.config.config import Config
+
+from ...exceptions import InvalidExportPath, ModelNotFound
+from ._repository import _AbstractRepository, _CustomDecoder, _CustomEncoder
+
+
+class _FileSystemRepository(_AbstractRepository[ModelType, Entity]):
+    """
+    Holds common methods to be used and extended when the need for saving
+    dataclasses as JSON files in local storage emerges.
+
+    Some lines have type: ignore because MyPy won't recognize some generic attributes. This
+    should be revised in the future.
+
+    Attributes:
+        model (ModelType): Generic dataclass.
+        dir_name (str): Folder that will hold the files for this dataclass model.
+    """
+
+    def __init__(self, model: Type[ModelType], dir_name: str):
+        self.model = model
+        self._dir_name = dir_name
+
+    @property
+    def dir_path(self):
+        return self._storage_folder / self._dir_name
+
+    @property
+    def _storage_folder(self) -> pathlib.Path:
+        return pathlib.Path(Config.global_config.storage_folder)
+
+    def _save(self, entity: Entity):
+        self.__create_directory_if_not_exists()
+        # TODO: implement _from_entity on models
+        model = self.model._from_entity(entity)
+        self.__get_model_filepath(model.id).write_text(
+            json.dumps(model.to_dict(), ensure_ascii=False, indent=0, cls=_CustomEncoder, check_circular=False)
+        )
+
+    def _get(self, filepath: pathlib.Path) -> Json:
+        with pathlib.Path(filepath).open(encoding="UTF-8") as source:
+            return json.load(source)
+
+    def load(self, model_id: str) -> Entity:
+        try:
+            model = self.model(**self._get(self.__get_model_filepath(model_id)))
+            return model._to_entity()
+        except FileNotFoundError:
+            raise ModelNotFound(str(self.dir_path), model_id)
+
+    def _load_all(self, version_number: Optional[str] = None) -> List[Entity]:
+        """
+        Load all entities from a specific version.
+        """
+        from ..._version._version_manager_factory import _VersionManagerFactory
+
+        version_number = _VersionManagerFactory._build_manager()._replace_version_number(version_number)
+
+        r = []
+        try:
+            for f in self.dir_path.iterdir():
+                if entity := self.model._to_entity(self.model(**self._get(f))):
```

Comment:
this can be a huge hit to the performance if we try to load all files. I think Florian did a very neat trick in the function `__match_file_and_get_entity` the current version of file `src/taipy/core/_repository/_filesystem_repository.py` where you read the file as a text file, look for the "data" in the form of text, if it exists in the file, read it as an entity. It still requires us to load all files but not as a taipy entity but as texts, do you think this will work in this case?

Reply:
I've added the logic of first loading the file contents and filtering before converting to a entity. I've remove some of logic that handled version numbers, but other than that it should behave very closely to the old repository. Of course we can always test after the FSRepo is done.